### DISCUSSION TITLE
How can I use the 'non-root' feature？

### DISCUSSION THREAD

**Discussion by qqliuxiaoran** [acassen/keepalived#2325] [Q&A]
Hello! @pqarmitage. Your reply in the previous discussion was very helpful to me
It's me again... XD

I notice that you plus new feature in v2.2.7(maybe even earlier) https://www.keepalived.org/release-notes/Release-2.2.7.html
```
global: Don’t assume running as user root.
systemd: Add keepalived-non-root.service systemd service file. keepalived-non-root.service allows keepalived to be run as a non root user, but with specific added capabilities to allow all the functionality that keepalived needs.
```
I searched for relevant information as much as possible, but couldn't find. 
Can you give me some documentation or suggestions on how to use this feature?

**Comment by pqarmitage** [comment]
@qqliuxiaoran I have absolutely no idea how to run keepalived as a non root user; it was something someone requested, and I saw i was possible to build keepalived to permit it to run as a non root user.

There are all sorts of permissions that need to be granted to keepalived, such as changing network configuration, so that keepalived has the permissions it needs to run, without being root and thereby having all permissions.

There should be an issue, although I do not know its number, where the feature was requested, and that may give more detail of the permissions needed.

**Reply by qqliuxiaoran** [reply]
OK!

**Comment by nser77** [comment]
This is the issue referenced above: https://github.com/acassen/keepalived/issues/2009
